memory_test.go

```diff
@@ -0,0 +1,447 @@
+			1 + SizeEmpty,
+		},
+		{
+			"reference to sparse array",
+			`x = []
+			x[5000] = "abcb";
+			x[5001] = "abc";
+			checkMem()
+			y = x;
+			// len("y") + reference to array
+			checkMem()`,
+			1 + SizeEmpty,
+		},
+		{
+			"Date object",
+			`
+			d1 = new Date();
+			checkMem();
+			d2 = new Date();
+			checkMem()
+			`,
+			// len("d2") + size of msec + reference to visited base object + base object prototype reference
+			2 + SizeNumber + SizeEmpty + SizeEmpty,
+		},
+		{
+			"Empty object",
+			`
+			checkMem();
+			o = {}
+			checkMem()
+			`,
+			// len("o") + object's starting size + reference to prototype
+			1 + SizeEmpty + SizeNumber,
+		},
+		{
+			"Map",
+			`var m = new Map();
+			m.set("a", 1);
+			checkMem();
+			m.set("abc", {"a":10, "b":"1234"});
+			checkMem();`,
+			3 + // "abc"
+				SizeEmpty + SizeEmpty + // outer object + reference to its prototype
+				(1 + SizeNumber) + // "a" and number
+				(1 + 4), // "b" and string
+		},
+		{
+			"Proxy",
+			`var target = {
+				message1: "hello",
+				message2: "everyone"
+			};
+			
+			var handler = {
+				get: function(target, prop, receiver) {
+					return "world";
+				}
+			};
+			var proxy1 = new Proxy(target, handler);
+			
+			checkMem();
+			proxy2 = new Proxy(target, handler);
+			checkMem();
+			`,
+			6 + // "proxy2"
+				SizeEmpty + // proxy overhead
+				SizeEmpty + SizeEmpty + // base object + prototype
+				SizeEmpty + // target object reference
+				SizeEmpty, // handler object reference
+		},
+		{
+			"String",
+			`str1 = new String("hi")
+
+			checkMem();
+			str2 = new String("hello")
+			checkMem();
+			`,
+			4 + // "str2"
+				5 + // "hello"
+				SizeEmpty + SizeEmpty + // base object + prototype
+				6 + SizeNumber, // "length" + number
+		},
+		{
+			"Typed array",
+			`var ta = new Uint8Array(1);
+			checkMem();
+			ta2 = new Uint8Array([1, 2, 3, 4]),
+			checkMem();
+			`,
+			3 + // "ta2"
+				SizeEmpty + // typed array overhead
+				SizeEmpty + SizeEmpty + // base object + prototype
+				4 + SizeEmpty + SizeEmpty + // array buffer data +  base object + prototype
+				SizeEmpty, // default constructor
+		},
+		{
+			"ArrayBuffer",
+			`var buffer = new ArrayBuffer(16);
+
+			checkMem();
+			buffer2 = new ArrayBuffer(16);
+			checkMem();`,
+			7 + // "buffer2"
+				16 + // data size
+				SizeEmpty + SizeEmpty, // base object + prototype
+		},
+		{
+			"DataView",
+			`var buffer = new ArrayBuffer(16);
+			var view = new DataView(buffer, 0);
+			var buffer2 = new ArrayBuffer(16);
+
+			checkMem();
+			view2 = new DataView(buffer2, 0);
+			checkMem();`,
+			5 + // "view2"
+				SizeEmpty + // DataView overhead
+				SizeEmpty + SizeEmpty + // base object + prototype
+				SizeEmpty, // array buffer reference
+		},
+		{
+			"Number",
+			`num1 = new Number("1")
+
+			checkMem();
+			num2 = new Number("2")
+			checkMem();`,
+			4 + // "num2"
+				SizeNumber +
+				SizeEmpty + SizeEmpty, // base object + prototype
+		},
+		// {
```

Comment:
We had a chat with Andy about this test failing and decided it was ok to comment out for the time being. Just wanted to call it out!